oap-server/server-storage-plugin/storage-elasticsearch-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/elasticsearch/query/MetadataQueryEsDAO.java

```diff
@@ -153,10 +153,12 @@ public List<ServiceInstance> listInstances(Duration duration,
         final String index =
             IndexController.LogicIndicesRegister.getPhysicalTableName(InstanceTraffic.INDEX_NAME);
 
-        final long minuteTimeBucket = TimeBucket.getMinuteTimeBucket(duration.getStartTimestamp());
+        final long startMinuteTimeBucket = TimeBucket.getMinuteTimeBucket(duration.getStartTimestamp());
+        final long endMinuteTimeBucket = TimeBucket.getMinuteTimeBucket(duration.getEndTimestamp());
         final BoolQueryBuilder query =
             Query.bool()
-                 .must(Query.range(InstanceTraffic.LAST_PING_TIME_BUCKET).gte(minuteTimeBucket))
+                 .must(Query.range(InstanceTraffic.LAST_PING_TIME_BUCKET).gte(startMinuteTimeBucket))
+                 .must(Query.range(InstanceTraffic.TIME_BUCKET).lt(endMinuteTimeBucket))
```

Comment:
According to using two conditions now, we should add a short-circuit logic in `MetadataQueryService#listInstances` to check `endMinuteTimeBucket` > `startMinuteTimeBucket`. 

Otherwise, we could query the instance list from illegal start and end timestamps.

Reply:
Currently, it appears that GraphQL queries are missing this validation logic. I believe the validation logic should be added in a common place for global validation.

Reply:
In most cases, I think it would get the empty list or the query doesn't care about the range. But for this specific query, you still could get a not-null list.

Reply:
We don't need to report an error when these illegal conditions are applied. I just suggest avoiding the not-null list. That is too strange.

Reply:
I strongly agree with your perspective. I have now implemented the short-circuit logic as you suggested.